azure_monitor/src/azure_monitor/sdk/auto_collection/metrics_span_processor.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+import logging
+
+from opentelemetry.trace import SpanKind
+from opentelemetry.sdk.trace import Span, SpanProcessor
+
+from azure_monitor.export import convert_span_to_envelope
+
+logger = logging.getLogger(__name__)
+
+
+class AzureMetricsSpanProcessor(SpanProcessor):
+    """AzureMetricsSpanProcessor is an implementation of `SpanProcessor` used
+    to generate Azure specific metrics, including dependencies/requests rate, average duration
+    and failed dependencies/requests.
+    """
+
+    def __init__(self, exporter=None):
+        self._exporter = exporter
```

Comment:
I don't think exporter should be an optional. Since it an `Azure` metrics span processor, we should instantiate the metrics exporter.

Reply:
Exporter removed from here, this was to have a reference to LiveMetricsExporter to be able to send documents to QuickPulse, updated the code to use the span processor in the exporter isntead